controllers/storagecluster/exporter.go

```diff
@@ -209,9 +263,496 @@ func CreateOrUpdateServiceMonitor(r *StorageClusterReconciler, instance *ocsv1.S
+		},
+	}
+	currentServiceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      expectedServiceAccount.Name,
+			Namespace: expectedServiceAccount.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, currentServiceAccount, func() error {
+		if len(currentServiceAccount.OwnerReferences) == 0 {
+			currentServiceAccount.OwnerReferences = make([]metav1.OwnerReference, 1)
+		}
+		currentOwner := &currentServiceAccount.OwnerReferences[0]
+		// if any of the ownerref criteria is not satisfied,
+		// set to the expected SA owenerref
+		if currentOwner.APIVersion == "" || currentOwner.Kind == "" ||
+			currentOwner.Name == "" || currentOwner.UID == "" {
+			currentServiceAccount.OwnerReferences[0] = expectedServiceAccount.OwnerReferences[0]
+		}
+		return nil
+	})
+	return err
+}
+
+// updateMetricsExporterClusterRoleBindings function updates the
+// cluster level rolebindings for the metrics exporter in this namespace
+func updateMetricsExporterClusterRoleBindings(ctx context.Context,
+	r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	currentCRB := &rbacv1.ClusterRoleBinding{ObjectMeta: metav1.ObjectMeta{
+		Name: metricsExporterName,
+	}}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, currentCRB, func() error {
+		currentCRB.RoleRef.Name = metricsExporterName
+		currentCRB.RoleRef.Kind = "ClusterRole"
+		currentCRB.RoleRef.APIGroup = "rbac.authorization.k8s.io"
+		subjectNotFound := true
+		expectedSubject := rbacv1.Subject{
+			Kind:      "ServiceAccount",
+			Name:      metricsExporterName,
+			Namespace: instance.Namespace,
+		}
+		for _, subject := range currentCRB.Subjects {
+			if subject.Kind == expectedSubject.Kind &&
+				subject.Name == expectedSubject.Name &&
+				subject.Namespace == expectedSubject.Namespace {
+				subjectNotFound = false
+				break
+			}
+		}
+		if subjectNotFound {
+			currentCRB.Subjects = append(currentCRB.Subjects, expectedSubject)
+		}
+		return nil
+	})
+	return err
+}
+
+func createMetricsExporterConfigMap(ctx context.Context,
+	r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	configMapName := "ocs-metrics-exporter-ceph-conf"
+	currentConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: instance.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, currentConfigMap, func() error {
+		currentConfigMap.Data = map[string]string{
+			"ceph.conf": `
+[global]
+auth_cluster_required = cephx
+auth_service_required = cephx
+auth_client_required = cephx
+`,
+			// keyring is a required key and its value should be empty
+			"keyring": "",
+		}
+		return nil
+	})
+	return err
+}
+
+const metricsExporterClusterRoleJSON = `
+{
+	"apiVersion":"rbac.authorization.k8s.io/v1",
+	"kind":"ClusterRole",
+	"metadata":{"name":"ocs-metrics-exporter"},
+	"rules":[
+		{
+			"apiGroups":["monitoring.coreos.com"],
+			"resources":["*"],
+			"verbs":["*"]
```

Comment:
Do not use "*". List each value explicitly.

Reply:
Done. Moved the `monitoring.coreos.com` apigroup to `Role` part as all the resources in it are namespace based.